H2MLauncher.UI/MainWindow.xaml.cs

```diff
@@ -1,29 +1,67 @@
 ﻿using System.ComponentModel;
+using System.Runtime.InteropServices;
 using System.Windows;
 using System.Windows.Controls;
 using System.Windows.Data;
 using System.Windows.Input;
+using System.Windows.Threading;
 
 using H2MLauncher.Core.Models;
 using H2MLauncher.Core.ViewModels;
+using H2MLauncher.UI.Model;
 
 namespace H2MLauncher.UI
 {
     public partial class MainWindow : Window
     {
         private readonly ICollectionView _collectionView;
         private readonly ICollectionView _collectionViewFavorites;
-
         private readonly ServerBrowserViewModel _viewModel;
+        private IntPtr _targetWindowHandle;
+        private bool _isOverlayVisible = true;
+        private bool _overlayHiddenByUser = false;
+        private GlobalKeyboardHook _globalKeyboardHook;
+
+        // Import Windows API methods for interacting with windows
+
+        [DllImport("user32.dll", SetLastError = true)]
+        private static extern IntPtr FindWindow(string lpClassName, string lpWindowName);
```

Comment:
We have these methods in the `H2MCommunicationService` It might be useful/better if we were able to reuse it. @tobibodamer any ideas?